### PR TITLE
feat: TypeScript for server

### PR DESCRIPTION
**Pull request by awave1**
Initial TypeScript support for Spotifyre backend.

This PR:
- Adds support for `ts-node`
- Translates the files from `.js` to `.ts`
- Adds partial typing support

### CODE REVIEW

File: server/src/routes/users.ts [hunk 1]
```diff
@@ -20,19 +21,6 @@ app.get('/get-curators/:query', (req, res) => {
     });
 });
 
-app.get('/getType', (req, res) => {
```
Comment (awave1) [inline server/src/routes/users.ts:23]: this endpoint wasn't used anywhere, so I just removed it

File: server/src/actions/types.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  id: string;
+  name: string;
+  artist: string;
+  albumArt: string;
+};
+
+export type SuggestedByUserInfo = {
+  id: string;
+  name: string;
+};
```
Comment (natekchua) [inline server/src/actions/types.ts:51]: newline

File: server/src/db/sql.ts [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
+import pg, { Defaults, PoolConfig } from 'pg';
 
 // Database config
-const config = {
+const config: PoolConfig & Defaults = {
   user: process.env.DB_user,
   database: process.env.DB,
   password: process.env.DB_password,
   host: process.env.DB_host,
-  // ssl: { rejectUnauthorized: false }, // uncomment in local dev environment
```
Comment (natekchua) [inline server/src/db/sql.ts:10]: can you leave this line please

File: server/src/db/sql.ts [hunk 4]
```diff
@@ -1,19 +1,24 @@
...
   port: 5432,
   poolSize: 5,
   poolIdleTimeout: 30000,
   reapIntervalMillis: 10000
 };
 
+if (process.env.NODE_ENV === "development") {
```
Comment (natekchua) [inline server/src/db/sql.ts:16]: single quotes

File: server/src/db/types.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export type DatabaseConfig = {
+  user: string;
+  database: string;
+  password: string;
+  host: string;
+  port: number;
+  poolSize: number;
+}
```
Comment (natekchua) [inline server/src/db/types.ts:8]: newline

File: server/src/models/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export * from './user';
```
Comment (natekchua) [inline server/src/models/index.ts:1]: newline

File: server/src/models/user.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+export type User = SpotifyApi.CurrentUsersProfileResponse;
+
+export type UserImage = {
+  url: string;
+}
+
+export type Followers = {
+  total: number;
+}
```
Comment (natekchua) [inline server/src/models/user.ts:9]: newline

File: server/src/routes/index.ts [hunk 8]
```diff
@@ -317,19 +316,17 @@ app.post('/api/search-for-playlists', async (req, res) => {
...
+    if (data?.body?.items) {
+      res.send({ topTracks: data?.body?.items });
+    }
+  } catch (err) {
+    console.log('Something went wrong!', err);
+  }
 });
 
-module.exports = app;
+export default app;
```
Comment (natekchua) [inline server/src/routes/index.ts:332]: newline

File: server/src/actions/actions.ts [hunk 9]
```diff
@@ -1,10 +1,12 @@
-const { SQL } = require('../db/sql.js');
+import { User } from '../models';
+import { SQL } from '../db/sql';
+import { AddSongSuggestionParams, AddUserParams, RemoveSongSuggestionParams, TokenData, UpdateUserParams } from './types';
```
Comment (natekchua) [inline server/src/actions/actions.ts:3]: should we newline each import here? 
`import {
   AddSongSuggestionParams,\n
   AddUserParams,\n
   RemoveSongSuggestionParams,\n
   TokenData,\n
   UpdateUserParams\n
} from './types';
`

File: server/src/db/sql.ts (hunk 3 above)
Comment (raksdewji) [reply server/src/db/sql.ts:10]: Don't need this anymore, using the ``process.env.NODE_ENV === "development"`` on line 16

File: server/src/actions/actions.ts (hunk 9 above)
Comment (awave1) [reply server/src/actions/actions.ts:3]: ideally, yea, that's what prettier would do, but eslint --fix decided to keep this

**Review by raksdewji — APPROVED** [review]

**Review by natekchua — APPROVED** [review]

**Pull request merged**